internal/pkg/client/oci/oci.go

```diff
@@ -114,17 +115,22 @@ func ImageSHA(ctx context.Context, uri string, sys *types.SystemContext) (string
 	return calculateRefHash(ctx, ref, sys)
 }
 
-func calculateRefHash(ctx context.Context, ref types.ImageReference, sys *types.SystemContext) (string, error) {
+func calculateRefHash(ctx context.Context, ref types.ImageReference, sys *types.SystemContext) (hash string, err error) {
 	source, err := ref.NewImageSource(ctx, sys)
 	if err != nil {
 		return "", err
 	}
+	defer func() {
+		if closeErr := source.Close(); closeErr != nil {
+			err = errors.Wrapf(err, " (src: %v)", closeErr)
```

Comment:
I'm trying to picture this from the point of view of the user. The user will see:

    Something something something (src: something something): something something

The extra space before ` (src: %v)` is something that the upper layers should be dealing with (as part of the leading "Something something something"). The additional ": something something" is the result of calling Error() on this new error (and the "soemthing something" will be the from the error returned by `ref.NewImageSource`.